controllers/machinedeletionremediation_controller.go

```diff
@@ -149,6 +149,36 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 
 	log.Info("node-associated machine found", "node", remediation.Name, "machine", machine.GetName())
 
+	// Detect if Node name is expected to change after Machine deletion given
+	// the Platform type. In case of BareMetal platform, the name is NOT
+	// expected to change, for other providers, instead, the name is expected
+	// to change.
+	// The providerID contains the platform type as prefix (e.g. baremetal:///...)
+	var providerID string
+	if providerID, err = getMachineProviderID(machine); err != nil {
+		log.Error(err, "could not get providerID")
+		return ctrl.Result{}, err
```

Comment:
I understand when we get an error here, there is a serious issue with the machine resource. Is there any chance that it will be resolved? If not, may don't requeue?

Reply:
indeed it is a quite serious problem, I was thinking that getting a new Machine from the cluster in the next try could fix it. However, if it is unexpected, I can change this